subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/platforms/JavaPlatformResolveIntegrationTest.groovy

```diff
@@ -132,7 +132,7 @@ class JavaPlatformResolveIntegrationTest extends AbstractHttpDependencyResolutio
         }
     }
 
-    def "reasonable behavior when using a regular project dependency instead of a platform dependency"() {
+    def "fails when using a regular project dependency instead of a platform dependency"() {
```

Comment:
I think it is correct that this fails. You should have tho use `platform(...)`. Otherwise it is also confusing that it locally works without `platform(...)`, but not for BOMs.